### ISSUE TITLE
download versions

### ISSUE THREAD

**Issue by ghost** [eggheads/eggdrop#427]
git has: 1.8.1 [sub -03]
geteggdrop has 1.8.1 [final]
eggheads has rc1

Why aren't these unified? (git should have the same build [if not NEWER] than a posted RC; in this case, it's OLDER!

**Comment by thommey** [comment]
Hm. Where are you getting those version numbers from?
Our latest stable release is 1.8.1 and our current RC is 1.8.2rc1.

The main website http://eggheads.org has buttons on the main site and links on http://edward.eggheads.org/downloads for both of these.

The github releases page on https://github.com/eggheads/eggdrop/releases has 1.8.1 and 1.8.2rc1 as well with 1.8.2rc1 marked as release candidate and 1.8.1 as latest stable.

The git develop branch is $::version "1.8.2+alpha", $::numversion 1080200. The release/1.8.2 branch has 1.8.2rc1 specifically.

Can you let me know how you get those version numbers, so I can reproduce it? Maybe our homepage caches too aggressively and you don't see the 1.8.2 version. Not sure about what could go wrong on git(hub) yet.

EDIT: Oh, and geteggdrop.com having 1.8.1 as latest stable is as intended. It will be updated once 1.8.2 becomes stable.

**Comment by vanosg** [comment]
Looking at version.h, the git develop branch has 1.8.2-alpha, as is expected. What are you looking at?
Geteggdrop only hosts stable code, that too is correct. Thanks.


Sent from my Verizon 4G LTE smartphone

-------- Original message --------

**Comment by ghost** [comment]
eggheads.org shows a link to 1.8.2+RC1.
geteggdrop.com shows 1.8.1
github shows 1.8.2+alpha

Full re-compiles of each set  (version.h & version reports when the bot is running) matches what's displayed.

I get that geteggdrop would only house the stable releases; doesn't explain why the other two don't match.

**Comment by vanosg** [comment]
The link on eggheads.org is to a packaged release tarball for the rc, develop is the ongoing developmental branch. We don't place tags like stable or rc in the develop branch. You can look at the release/1.8.2 branch to find the rc1 tag, or by clicking the 'releases' link on GitHub.

**Comment by thommey** [comment]
Because it might be useful for reference I'll outline it a bit more detailed:

Previously, on CVS, we had a linear version history:
1.6.19->1.6.20rc1->1.6.20rc2->1.6.20->1.6.21rc1->... and whenever you checked out cvs, you got whatever the latest RC/release is + newest modifications.

Now, in git, we can (and should) use multiple branches to be able to develop multiple versions at a time:
```
--o--o--o--C--x--x--x--x <- develop branch (C = last feature included in v1.8.2rc1)
            \-o--A--o--B <- release/1.8.2 branch (A = v1.8.2rc1; B = v1.8.2 stable)
--S <- release/1.8.1 branch and stable/1.8 branch (S = v1.8.1, latest stable 1.8.x)
```
o, x = commit, C = commit, A/B = commits with release tags

So, after v1.8.2rc1 is released or branched off, we can start developing on the develop branch for v1.8.3. And that is what you get when you checkout the develop branch, our start of work for 1.8.3.
Of course, patches between 1.8.2rc1 and 1.8.2rc2 get integrated into it (or vice versa), but at that point you have three choices:
1) Check out the **develop** branch and get our v1.8.3 draft
2) Check out the **release/1.8.2** branch and get latest v1.8.2 release (rc1, rc2, stable)
3) Check out the **stable/1.8** branch to get our latest v1.8 stable release (v1.8.1 right now)

In addition, you can also get our release tarballs (RC or stable) from https://github.com/eggheads/eggdrop/releases

Now to the actual version number it shows on the develop branch. As I said above, the develop branch is our start of v1.8.3 as soon as the release/1.8.2 branch is created, so it SHOULD show v1.8.3+alpha (or whatever suffix is suitable). We haven't quite gotten the release process down 100% yet (mostly because of the much more complicated changelog generation because of the branches).

So, thanks for pointing out that develop still says v1.8.2+alpha, we'll have to do some discussing on what we should call develop right now, but it should reflect that it's the 1.8.3 draft, not 1.8.2. I hope that clarifies things a bit (for everyone that reads this or that I point to this ;) )
